process/serialize.go

```diff
@@ -0,0 +1,237 @@
+	for _, key := range keys {
+		ser.Add(key, data[key])
+	}
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("2", data.GetStringConst())
+	ser.AddFloat("3", data.GetDoubleConst())
+	ser.AddBool("4", data.GetBoolConst())
+	ser.Add("5", data.GetRef())
+	ser.Add("6", data.GetList())
+	ser.Add("7", data.GetMap())
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_Map) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("1", mapProcessNodeMapOutput(data.Outputs))
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_List) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("1", processNodeMapOutputs(data.Outputs))
+	return ser.HashSerialize()
+}
+
+type processNodeMapOutputs []*Process_Node_Map_Output
+
+// HashSerialize returns the hashserialized string of this type
+func (data processNodeMapOutputs) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	for i, value := range data {
+		ser.Add(strconv.Itoa(i), value)
+	}
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_Reference) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("1", data.NodeKey)
+	ser.Add("2", data.Path)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Map_Output_Reference_Path) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddString("1", data.GetKey())
+	ser.AddInt("2", int(data.GetIndex()))
+	ser.Add("3", data.Path)
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Process_Node_Filter) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("2", processNodeFilterConditions(data.Conditions))
+	return ser.HashSerialize()
+}
+
+type processNodeFilterConditions []Process_Node_Filter_Condition
+
+// HashSerialize returns the hashserialized string of this type
+func (data processNodeFilterConditions) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	for i, value := range data {
+		ser.Add(strconv.Itoa(i), value)
+	}
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data Process_Node_Filter_Condition) HashSerialize() string {
+	ser := hashserializer.New()
+	ser.AddString("1", data.Key)
+	ser.AddInt("2", int(data.Predicate))
```

Comment:
Value of the condition is missing here and should definitely be part of the hash